src/main/java/com/palominolabs/metrics/newrelic/CustomMetricAttributeFilter.java

```diff
@@ -0,0 +1,466 @@
+
+import com.codahale.metrics.*;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Supplier;
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+
+/**
+ * Convenience implementation of {@link MetricAttributeFilter} with grain level customization.
+ * Two constructors are provided:<br>
+ * # first which receives a {@link Table} indexed by metricName and NewRelic metric (@see NewRelicMetric).
+ * For example:
+ * <pre>
+ *     {@code 
+ *      Table<String, NewRelicMetric, Boolean> metricConfig = ImmutableTable.<String, NewRelicMetric, Boolean>builder()
+ *          .put("metricName1", NewRelicMetric.TIMER_MAX, true)
+ *          .put("metricName1", NewRelicMetric.TIMER_MIN, false)
+ *          .put("metricName2", NewRelicMetric.COUNTER_COUNT, true)
+ *          .build();
+ *
+ *      new CustomMetricAttributeFilter(metricConfig, new AllDisabledMetricAttributeFilter());
+ *     }
+ * </pre>
+ * # second which receives a {@link URI} to a <i>yaml</i> file where metrics configuration is defined in the folowwing format:
+ * <pre>
+ *     metricName1:
+ *          TIMER_MAX: true
+ *          TIMER_MIN: false
+ *     metricName2:
+ *          COUNTER_COUNT: true
+ * </pre>
+ *
+ * Both constructors receives a {@link MetricAttributeFilter} which will be used as fallback for all metrics not specified in configuration.
+ * In case <i>fallback</i> is null, {@link AllDisabledMetricAttributeFilter} will be used.
+ */
+public class CustomMetricAttributeFilter implements MetricAttributeFilter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CustomMetricAttributeFilter.class);
+
+    private Table<String, NewRelicMetric, Boolean> enabledMetrics;
+    private MetricAttributeFilter fallback;
+
+    /**
+     * @param metricConfig table containing metrics config
+     * @param fallback to be used when there metrics config has no entry for the metric to be reported
+     */
+    public CustomMetricAttributeFilter(Table<String, NewRelicMetric, Boolean> metricConfig, @Nullable MetricAttributeFilter fallback) {
+        Preconditions.checkArgument(metricConfig != null, "metricConfig cannot be null");
+        this.enabledMetrics = metricConfig;
+        this.fallback = getFallbackMetricFilter(fallback);
+    }
+
+    /**
+     * @param metricConfigURI URI to a yaml file containing metrics config
+     * @param fallback to be used when there metrics config has no entry for the metric to be reported
+     */
+    public CustomMetricAttributeFilter(URI metricConfigURI, @Nullable MetricAttributeFilter fallback) {
```

Comment:
What do you think of receiving a "TableSupplier"? We could include 2 implementations, YamlTableSupplier and Raw/DefaultTableSupplier? 
Otherwise, we can move foward with your proposal

http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Supplier.html